jcommune-view/jcommune-web-controller/src/main/java/org/jtalks/jcommune/web/listeners/SessionSetupListener.java

```diff
@@ -58,6 +72,43 @@ public void sessionCreated(HttpSessionEvent se) {
      */
     @Override
     public void sessionDestroyed(HttpSessionEvent se) {
-        //noop
+        LocationService locationService;
+        ServletContext context = se.getSession().getServletContext();
+        Object principal = getPrincipalFromSession(se.getSession());
+        if (principal != null) {
+            try {
+                locationService = getBeanFormApplicationContext(context, LocationService.class, "locationService");
```

Comment:
Is it possible to retrieve that bean only once? I'm not sure how performant this operation is in Spring..